fix(web-components): use lib replacement for web types

radium-v opened the pull request:
## Previous Behavior

The type declaration files generated for the `@fluentui/web-components` package are incorrectly including reference tags (`/// <reference types="web" />`), which creates issues for projects which use `@fluentui/web-components` as a dev dependency.

Many new browser features we're using (like the Popover API) are sometimes too new to rely on the DOM types provided with the version of TypeScript in the repository, but these types are expected to be included with TypeScript over time, so we don't want to include them as a `types` reference.

## New Behavior

This PR uses lib replacement to replace the built-in `@typescript/lib-dom` package with `@types/web`, which prevents `/// <reference types="web" />` tags from being included in declaration files. The replaced types package is only included in the `web-components/package.json` and is added to the `nohoist` list to prevent it from being referenced outside of the `web-components` package.

Additional changes:
* Fixed a minor type issue with the `MatchMediaStyleSheetBehavior` utility class.

fabricteam commented:
## 🕵 fluentui-web-components-v3 No visual regressions between this PR and main
<div id="vrtCommentfluent-ui_VRT_Pipeline_web-components"/>

fabricteam commented:
## [Perf Analysis (`@fluentui/react-components`)](https://github.com/microsoft/fluentui/wiki/Perf-Testing)

  <table>
  <tr>
    <th>Scenario</th>
    <th>Render type</th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">Master Ticks</a>
    </th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">PR Ticks</a>
    </th>
    <th>Iterations</th>
    <th>Status</th>
  </tr><tr>
            <td>FluentProviderWithTheme</td>
            <td>virtual-rerender</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/32129/perf-test-react-components/FluentProviderWithTheme-virtual-rerender_base.html">33</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/32129/perf-test-react-components/FluentProviderWithTheme-virtual-rerender.html">33</a></td>
            <td>10</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/32129/perf-test-react-components/FluentProviderWithTheme-virtual-rerender.regression.txt">Possible regression</a></td>
           </tr></table><details><summary>All results</summary><p>
  <table>
  <tr>
    <th>Scenario</th>
    <th>Render type</th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">Master Ticks</a>
    </th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">PR Ticks</a>
    </th>
    <th>Iterations</th>
    <th>Status</th>
  </tr><tr>
            <td>Avatar</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/32129/perf-test-react-components/Avatar-mount_base.html">648</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/32129/perf-test-react-components/Avatar-mount.html">630</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Button</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/32129/perf-test-react-components/Button-mount_base.html">308</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/32129/perf-test-react-components/Button-mount.html">312</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Field</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/32129/perf-test-react-components/Field-mount_base.html">1108</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/32129/perf-test-react-components/Field-mount.html">1126</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProvider</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/32129/perf-test-react-components/FluentProvider-mount_base.html">721</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/32129/perf-test-react-components/FluentProvider-mount.html">719</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/32129/perf-test-react-components/FluentProviderWithTheme-mount_base.html">84</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/32129/perf-test-react-components/FluentProviderWithTheme-mount.html">79</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>virtual-rerender</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/32129/perf-test-react-components/FluentProviderWithTheme-virtual-rerender_base.html">33</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/32129/perf-test-react-components/FluentProviderWithTheme-virtual-rerender.html">33</a></td>
            <td>10</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/32129/perf-test-react-components/FluentProviderWithTheme-virtual-rerender.regression.txt">Possible regression</a></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>virtual-rerender-with-unmount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/32129/perf-test-react-components/FluentProviderWithTheme-virtual-rerender-with-unmount_base.html">80</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/32129/perf-test-react-components/FluentProviderWithTheme-virtual-rerender-with-unmount.html">86</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>MakeStyles</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/32129/perf-test-react-components/MakeStyles-mount_base.html">864</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/32129/perf-test-react-components/MakeStyles-mount.html">856</a></td>
            <td>50000</td>
            <td></td>
           </tr>
<tr>
            <td>Persona</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/32129/perf-test-react-components/Persona-mount_base.html">1802</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/32129/perf-test-react-components/Persona-mount.html">1733</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>SpinButton</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/32129/perf-test-react-components/SpinButton-mount_base.html">1409</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/32129/perf-test-react-components/SpinButton-mount.html">1407</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>SwatchPicker</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/32129/perf-test-react-components/SwatchPicker-mount_base.html">1680</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/32129/perf-test-react-components/SwatchPicker-mount.html">1653</a></td>
            <td>5000</td>
            <td></td>
           </tr></table></p></details>

<div id="perfComment9425"/>

fabricteam commented:
## 🕵 fluentuiv8 No visual regressions between this PR and main
<div id="vrtCommentfluent-ui_VRT_Pipeline_v8"/>

fabricteam commented:
## 📊 Bundle size report

✅ No changes found
<div id="bundleSizeComment9423"/>

fabricteam commented:
## 🕵 FluentUIV0 No visual regressions between this PR and main
<div id="vrtCommentfluent-ui_VRT_Pipeline_v0"/>

fabricteam commented:
## Perf Analysis (`@fluentui/react-northstar`)
<details><summary>Perf tests with no regressions</summary>

Scenario | Current PR Ticks | Baseline Ticks | Ratio
:--- | ---:| ---:| ---:
AccordionMinimalPerf.default | [91](https://fluentuipr.z22.web.core.windows.net/pull/32129/perf-test-northstar/AccordionMinimalPerf.default.html) | [78](https://fluentuipr.z22.web.core.windows.net/pull/32129/perf-test-northstar/AccordionMinimalPerf.default_base.html) | 1.17:1
ReactionMinimalPerf.default | [229](https://fluentuipr.z22.web.core.windows.net/pull/32129/perf-test-northstar/ReactionMinimalPerf.default.html) | [211](https://fluentuipr.z22.web.core.windows.net/pull/32129/perf-test-northstar/ReactionMinimalPerf.default_base.html) | 1.09:1
SkeletonMinimalPerf.default | [213](https://fluentuipr.z22.web.core.windows.net/pull/32129/perf-test-northstar/SkeletonMinimalPerf.default.html) | [198](https://fluentuipr.z22.web.core.windows.net/pull/32129/perf-test-northstar/SkeletonMinimalPerf.default_base.html) | 1.08:1
AvatarMinimalPerf.default | [118](https://fluentuipr.z22.web.core.windows.net/pull/32129/perf-test-northstar/AvatarMinimalPerf.default.html) | [110](https://fluentuipr.z22.web.core.windows.net/pull/32129/perf-test-northstar/AvatarMinimalPerf.default_base.html) | 1.07:1
GridMinimalPerf.default | [201](https://fluentuipr.z22.web.core.windows.net/pull/32129/perf-test-northstar/GridMinimalPerf.default.html) | [187](https://fluentuipr.z22.web.core.windows.net/pull/32129/perf-test-northstar/GridMinimalPerf.default_base.html) | 1.07:1
TableMinimalPerf.default | [246](https://fluentuipr.z22.web.core.windows.net/pull/32129/perf-test-northstar/TableMinimalPerf.default.html) | [230](https://fluentuipr.z22.web.core.windows.net/pull/32129/perf-test-northstar/TableMinimalPerf.default_base.html) | 1.07:1
StatusMinimalPerf.default | [411](https://fluentuipr.z22.web.core.windows.net/pull/32129/perf-test-northstar/StatusMinimalPerf.default.html) | [389](https://fluentuipr.z22.web.core.windows.net/pull/32129/perf-test-northstar/StatusMinimalPerf.default_base.html) | 1.06:1
TextAreaMinimalPerf.default | [305](https://fluentuipr.z22.web.core.windows.net/pull/32129/perf-test-northstar/TextAreaMinimalPerf.default.html) | [287](https://fluentuipr.z22.web.core.windows.net/pull/32129/perf-test-northstar/TextAreaMinimalPerf.default_base.html) | 1.06:1
AttachmentMinimalPerf.default | [89](https://fluentuipr.z22.web.core.windows.net/pull/32129/perf-test-northstar/AttachmentMinimalPerf.default.html) | [85](https://fluentuipr.z22.web.core.windows.net/pull/32129/perf-test-northstar/AttachmentMinimalPerf.default_base.html) | 1.05:1
ButtonMinimalPerf.default | [88](https://fluentuipr.z22.web.core.windows.net/pull/32129/perf-test-northstar/ButtonMinimalPerf.default.html) | [84](https://fluentuipr.z22.web.core.windows.net/pull/32129/perf-test-northstar/ButtonMinimalPerf.default_base.html) | 1.05:1
ListNestedPerf.default | [335](https://fluentuipr.z22.web.core.windows.net/pull/32129/perf-test-northstar/ListNestedPerf.default.html) | [320](https://fluentuipr.z22.web.core.windows.net/pull/32129/perf-test-northstar/ListNestedPerf.default_base.html) | 1.05:1
RadioGroupMinimalPerf.default | [275](https://fluentuipr.z22.web.core.windows.net/pull/32129/perf-test-northstar/RadioGroupMinimalPerf.default.html) | [262](https://fluentuipr.z22.web.core.windows.net/pull/32129/perf-test-northstar/RadioGroupMinimalPerf.default_base.html) | 1.05:1
RefMinimalPerf.default | [116](https://fluentuipr.z22.web.core.windows.net/pull/32129/perf-test-northstar/RefMinimalPerf.default.html) | [110](https://fluentuipr.z22.web.core.windows.net/pull/32129/perf-test-northstar/RefMinimalPerf.default_base.html) | 1.05:1
CarouselMinimalPerf.default | [266](https://fluentuipr.z22.web.core.windows.net/pull/32129/perf-test-northstar/CarouselMinimalPerf.default.html) | [255](https://fluentuipr.z22.web.core.windows.net/pull/32129/perf-test-northstar/CarouselMinimalPerf.default_base.html) | 1.04:1
CheckboxMinimalPerf.default | [1161](https://fluentuipr.z22.web.core.windows.net/pull/32129/perf-test-northstar/CheckboxMinimalPerf.default.html) | [1121](https://fluentuipr.z22.web.core.windows.net/pull/32129/perf-test-northstar/CheckboxMinimalPerf.default_base.html) | 1.04:1
DropdownManyItemsPerf.default | [405](https://fluentuipr.z22.web.core.windows.net/pull/32129/perf-test-northstar/DropdownManyItemsPerf.default.html) | [388](https://fluentuipr.z22.web.core.windows.net/pull/32129/perf-test-northstar/DropdownManyItemsPerf.default_base.html) | 1.04:1
HeaderSlotsPerf.default | [474](https://fluentuipr.z22.web.core.windows.net/pull/32129/perf-test-northstar/HeaderSlotsPerf.default.html) | [456](https://fluentuipr.z22.web.core.windows.net/pull/32129/perf-test-northstar/HeaderSlotsPerf.default_base.html) | 1.04:1
ImageMinimalPerf.default | [230](https://fluentuipr.z22.web.core.windows.net/pull/32129/perf-test-northstar/ImageMinimalPerf.default.html) | [222](https://fluentuipr.z22.web.core.windows.net/pull/32129/perf-test-northstar/ImageMinimalPerf.default_base.html) | 1.04:1
AnimationMinimalPerf.default | [306](https://fluentuipr.z22.web.core.windows.net/pull/32129/perf-test-northstar/AnimationMinimalPerf.default.html) | [298](https://fluentuipr.z22.web.core.windows.net/pull/32129/perf-test-northstar/AnimationMinimalPerf.default_base.html) | 1.03:1
AttachmentSlotsPerf.default | [658](https://fluentuipr.z22.web.core.windows.net/pull/32129/perf-test-northstar/AttachmentSlotsPerf.default.html) | [638](https://fluentuipr.z22.web.core.windows.net/pull/32129/perf-test-northstar/AttachmentSlotsPerf.default_base.html) | 1.03:1
ButtonOverridesMissPerf.default | [666](https://fluentuipr.z22.web.core.windows.net/pull/32129/perf-test-northstar/ButtonOverridesMissPerf.default.html) | [648](https://fluentuipr.z22.web.core.windows.net/pull/32129/perf-test-northstar/ButtonOverridesMissPerf.default_base.html) | 1.03:1
DialogMinimalPerf.default | [461](https://fluentuipr.z22.web.core.windows.net/pull/32129/perf-test-northstar/DialogMinimalPerf.default.html) | [447](https://fluentuipr.z22.web.core.windows.net/pull/32129/perf-test-northstar/DialogMinimalPerf.default_base.html) | 1.03:1
ItemLayoutMinimalPerf.default | [724](https://fluentuipr.z22.web.core.windows.net/pull/32129/perf-test-northstar/ItemLayoutMinimalPerf.default.html) | [701](https://fluentuipr.z22.web.core.windows.net/pull/32129/perf-test-northstar/ItemLayoutMinimalPerf.default_base.html) | 1.03:1
PortalMinimalPerf.default | [89](https://fluentuipr.z22.web.core.windows.net/pull/32129/perf-test-northstar/PortalMinimalPerf.default.html) | [86](https://fluentuipr.z22.web.core.windows.net/pull/32129/perf-test-northstar/PortalMinimalPerf.default_base.html) | 1.03:1
SliderMinimalPerf.default | [756](https://fluentuipr.z22.web.core.windows.net/pull/32129/perf-test-northstar/SliderMinimalPerf.default.html) | [737](https://fluentuipr.z22.web.core.windows.net/pull/32129/perf-test-northstar/SliderMinimalPerf.default_base.html) | 1.03:1
FormMinimalPerf.default | [226](https://fluentuipr.z22.web.core.windows.net/pull/32129/perf-test-northstar/FormMinimalPerf.default.html) | [222](https://fluentuipr.z22.web.core.windows.net/pull/32129/perf-test-northstar/FormMinimalPerf.default_base.html) | 1.02:1
HeaderMinimalPerf.default | [213](https://fluentuipr.z22.web.core.windows.net/pull/32129/perf-test-northstar/HeaderMinimalPerf.default.html) | [209](https://fluentuipr.z22.web.core.windows.net/pull/32129/perf-test-northstar/HeaderMinimalPerf.default_base.html) | 1.02:1
LayoutMinimalPerf.default | [201](https://fluentuipr.z22.web.core.windows.net/pull/32129/perf-test-northstar/LayoutMinimalPerf.default.html) | [198](https://fluentuipr.z22.web.core.windows.net/pull/32129/perf-test-northstar/LayoutMinimalPerf.default_base.html) | 1.02:1
ListWith60ListItems.default | [375](https://fluentuipr.z22.web.core.windows.net/pull/32129/perf-test-northstar/ListWith60ListItems.default.html) | [367](https://fluentuipr.z22.web.core.windows.net/pull/32129/perf-test-northstar/ListWith60ListItems.default_base.html) | 1.02:1
PopupMinimalPerf.default | [351](https://fluentuipr.z22.web.core.windows.net/pull/32129/perf-test-northstar/PopupMinimalPerf.default.html) | [343](https://fluentuipr.z22.web.core.windows.net/pull/32129/perf-test-northstar/PopupMinimalPerf.default_base.html) | 1.02:1
ProviderMinimalPerf.default | [205](https://fluentuipr.z22.web.core.windows.net/pull/32129/perf-test-northstar/ProviderMinimalPerf.default.html) | [200](https://fluentuipr.z22.web.core.windows.net/pull/32129/perf-test-northstar/ProviderMinimalPerf.default_base.html) | 1.02:1
TableManyItemsPerf.default | [1125](https://fluentuipr.z22.web.core.windows.net/pull/32129/perf-test-northstar/TableManyItemsPerf.default.html) | [1108](https://fluentuipr.z22.web.core.windows.net/pull/32129/perf-test-northstar/TableManyItemsPerf.default_base.html) | 1.02:1
CustomToolbarPrototype.default | [1504](https://fluentuipr.z22.web.core.windows.net/pull/32129/perf-test-northstar/CustomToolbarPrototype.default.html) | [1480](https://fluentuipr.z22.web.core.windows.net/pull/32129/perf-test-northstar/CustomToolbarPrototype.default_base.html) | 1.02:1
DropdownMinimalPerf.default | [1430](https://fluentuipr.z22.web.core.windows.net/pull/32129/perf-test-northstar/DropdownMinimalPerf.default.html) | [1419](https://fluentuipr.z22.web.core.windows.net/pull/32129/perf-test-northstar/DropdownMinimalPerf.default_base.html) | 1.01:1
ListCommonPerf.default | [394](https://fluentuipr.z22.web.core.windows.net/pull/32129/perf-test-northstar/ListCommonPerf.default.html) | [389](https://fluentuipr.z22.web.core.windows.net/pull/32129/perf-test-northstar/ListCommonPerf.default_base.html) | 1.01:1
TooltipMinimalPerf.default | [1311](https://fluentuipr.z22.web.core.windows.net/pull/32129/perf-test-northstar/TooltipMinimalPerf.default.html) | [1301](https://fluentuipr.z22.web.core.windows.net/pull/32129/perf-test-northstar/TooltipMinimalPerf.default_base.html) | 1.01:1
TreeMinimalPerf.default | [490](https://fluentuipr.z22.web.core.windows.net/pull/32129/perf-test-northstar/TreeMinimalPerf.default.html) | [487](https://fluentuipr.z22.web.core.windows.net/pull/32129/perf-test-northstar/TreeMinimalPerf.default_base.html) | 1.01:1
TreeWith60ListItems.default | [86](https://fluentuipr.z22.web.core.windows.net/pull/32129/perf-test-northstar/TreeWith60ListItems.default.html) | [85](https://fluentuipr.z22.web.core.windows.net/pull/32129/perf-test-northstar/TreeWith60ListItems.default_base.html) | 1.01:1
ButtonSlotsPerf.default | [317](https://fluentuipr.z22.web.core.windows.net/pull/32129/perf-test-northstar/ButtonSlotsPerf.default.html) | [316](https://fluentuipr.z22.web.core.windows.net/pull/32129/perf-test-northstar/ButtonSlotsPerf.default_base.html) | 1:1
ChatMinimalPerf.default | [441](https://fluentuipr.z22.web.core.windows.net/pull/32129/perf-test-northstar/ChatMinimalPerf.default.html) | [440](https://fluentuipr.z22.web.core.windows.net/pull/32129/perf-test-northstar/ChatMinimalPerf.default_base.html) | 1:1
DividerMinimalPerf.default | [209](https://fluentuipr.z22.web.core.windows.net/pull/32129/perf-test-northstar/DividerMinimalPerf.default.html) | [210](https://fluentuipr.z22.web.core.windows.net/pull/32129/perf-test-northstar/DividerMinimalPerf.default_base.html) | 1:1
MenuButtonMinimalPerf.default | [962](https://fluentuipr.z22.web.core.windows.net/pull/32129/perf-test-northstar/MenuButtonMinimalPerf.default.html) | [959](https://fluentuipr.z22.web.core.windows.net/pull/32129/perf-test-northstar/MenuButtonMinimalPerf.default_base.html) | 1:1
ChatDuplicateMessagesPerf.default | [150](https://fluentuipr.z22.web.core.windows.net/pull/32129/perf-test-northstar/ChatDuplicateMessagesPerf.default.html) | [152](https://fluentuipr.z22.web.core.windows.net/pull/32129/perf-test-northstar/ChatDuplicateMessagesPerf.default_base.html) | 0.99:1
DatepickerMinimalPerf.default | [3633](https://fluentuipr.z22.web.core.windows.net/pull/32129/perf-test-northstar/DatepickerMinimalPerf.default.html) | [3671](https://fluentuipr.z22.web.core.windows.net/pull/32129/perf-test-northstar/DatepickerMinimalPerf.default_base.html) | 0.99:1
EmbedMinimalPerf.default | [1872](https://fluentuipr.z22.web.core.windows.net/pull/32129/perf-test-northstar/EmbedMinimalPerf.default.html) | [1891](https://fluentuipr.z22.web.core.windows.net/pull/32129/perf-test-northstar/EmbedMinimalPerf.default_base.html) | 0.99:1
RosterPerf.default | [1581](https://fluentuipr.z22.web.core.windows.net/pull/32129/perf-test-northstar/RosterPerf.default.html) | [1596](https://fluentuipr.z22.web.core.windows.net/pull/32129/perf-test-northstar/RosterPerf.default_base.html) | 0.99:1
IconMinimalPerf.default | [395](https://fluentuipr.z22.web.core.windows.net/pull/32129/perf-test-northstar/IconMinimalPerf.default.html) | [397](https://fluentuipr.z22.web.core.windows.net/pull/32129/perf-test-northstar/IconMinimalPerf.default_base.html) | 0.99:1
BoxMinimalPerf.default | [187](https://fluentuipr.z22.web.core.windows.net/pull/32129/perf-test-northstar/BoxMinimalPerf.default.html) | [190](https://fluentuipr.z22.web.core.windows.net/pull/32129/perf-test-northstar/BoxMinimalPerf.default_base.html) | 0.98:1
LoaderMinimalPerf.default | [189](https://fluentuipr.z22.web.core.windows.net/pull/32129/perf-test-northstar/LoaderMinimalPerf.default.html) | [192](https://fluentuipr.z22.web.core.windows.net/pull/32129/perf-test-northstar/LoaderMinimalPerf.default_base.html) | 0.98:1
ProviderMergeThemesPerf.default | [637](https://fluentuipr.z22.web.core.windows.net/pull/32129/perf-test-northstar/ProviderMergeThemesPerf.default.html) | [652](https://fluentuipr.z22.web.core.windows.net/pull/32129/perf-test-northstar/ProviderMergeThemesPerf.default_base.html) | 0.98:1
SegmentMinimalPerf.default | [198](https://fluentuipr.z22.web.core.windows.net/pull/32129/perf-test-northstar/SegmentMinimalPerf.default.html) | [202](https://fluentuipr.z22.web.core.windows.net/pull/32129/perf-test-northstar/SegmentMinimalPerf.default_base.html) | 0.98:1
ToolbarMinimalPerf.default | [533](https://fluentuipr.z22.web.core.windows.net/pull/32129/perf-test-northstar/ToolbarMinimalPerf.default.html) | [546](https://fluentuipr.z22.web.core.windows.net/pull/32129/perf-test-northstar/ToolbarMinimalPerf.default_base.html) | 0.98:1
CardMinimalPerf.default | [303](https://fluentuipr.z22.web.core.windows.net/pull/32129/perf-test-northstar/CardMinimalPerf.default.html) | [311](https://fluentuipr.z22.web.core.windows.net/pull/32129/perf-test-northstar/CardMinimalPerf.default_base.html) | 0.97:1
InputMinimalPerf.default | [535](https://fluentuipr.z22.web.core.windows.net/pull/32129/perf-test-northstar/InputMinimalPerf.default.html) | [553](https://fluentuipr.z22.web.core.windows.net/pull/32129/perf-test-northstar/InputMinimalPerf.default_base.html) | 0.97:1
LabelMinimalPerf.default | [217](https://fluentuipr.z22.web.core.windows.net/pull/32129/perf-test-northstar/LabelMinimalPerf.default.html) | [224](https://fluentuipr.z22.web.core.windows.net/pull/32129/perf-test-northstar/LabelMinimalPerf.default_base.html) | 0.97:1
SplitButtonMinimalPerf.default | [2224](https://fluentuipr.z22.web.core.windows.net/pull/32129/perf-test-northstar/SplitButtonMinimalPerf.default.html) | [2285](https://fluentuipr.z22.web.core.windows.net/pull/32129/perf-test-northstar/SplitButtonMinimalPerf.default_base.html) | 0.97:1
TextMinimalPerf.default | [193](https://fluentuipr.z22.web.core.windows.net/pull/32129/perf-test-northstar/TextMinimalPerf.default.html) | [198](https://fluentuipr.z22.web.core.windows.net/pull/32129/perf-test-northstar/TextMinimalPerf.default_base.html) | 0.97:1
ChatWithPopoverPerf.default | [206](https://fluentuipr.z22.web.core.windows.net/pull/32129/perf-test-northstar/ChatWithPopoverPerf.default.html) | [214](https://fluentuipr.z22.web.core.windows.net/pull/32129/perf-test-northstar/ChatWithPopoverPerf.default_base.html) | 0.96:1
ListMinimalPerf.default | [301](https://fluentuipr.z22.web.core.windows.net/pull/32129/perf-test-northstar/ListMinimalPerf.default.html) | [313](https://fluentuipr.z22.web.core.windows.net/pull/32129/perf-test-northstar/ListMinimalPerf.default_base.html) | 0.96:1
VideoMinimalPerf.default | [452](https://fluentuipr.z22.web.core.windows.net/pull/32129/perf-test-northstar/VideoMinimalPerf.default.html) | [469](https://fluentuipr.z22.web.core.windows.net/pull/32129/perf-test-northstar/VideoMinimalPerf.default_base.html) | 0.96:1
AlertMinimalPerf.default | [162](https://fluentuipr.z22.web.core.windows.net/pull/32129/perf-test-northstar/AlertMinimalPerf.default.html) | [170](https://fluentuipr.z22.web.core.windows.net/pull/32129/perf-test-northstar/AlertMinimalPerf.default_base.html) | 0.95:1
FlexMinimalPerf.default | [153](https://fluentuipr.z22.web.core.windows.net/pull/32129/perf-test-northstar/FlexMinimalPerf.default.html) | [161](https://fluentuipr.z22.web.core.windows.net/pull/32129/perf-test-northstar/FlexMinimalPerf.default_base.html) | 0.95:1
MenuMinimalPerf.default | [482](https://fluentuipr.z22.web.core.windows.net/pull/32129/perf-test-northstar/MenuMinimalPerf.default.html) | [506](https://fluentuipr.z22.web.core.windows.net/pull/32129/perf-test-northstar/MenuMinimalPerf.default_base.html) | 0.95:1

</details><div id="perfComment9424"/>

fabricteam commented:
## 🕵 fluentuiv9 No visual regressions between this PR and main
<div id="vrtCommentfluent-ui_VRT_Pipeline_v9"/>

fabricteam commented:
## [Perf Analysis (`@fluentui/react`)](https://github.com/microsoft/fluentui/wiki/Perf-Testing)
<p>No significant results to display.</p><details><summary>All results</summary><p>
  <table>
  <tr>
    <th>Scenario</th>
    <th>Render type</th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">Master Ticks</a>
    </th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">PR Ticks</a>
    </th>
    <th>Iterations</th>
    <th>Status</th>
  </tr><tr>
            <td>BaseButton</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/32129/perf-test/BaseButton-mount_base.html">640</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/32129/perf-test/BaseButton-mount.html">641</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Breadcrumb</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/32129/perf-test/Breadcrumb-mount_base.html">1681</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/32129/perf-test/Breadcrumb-mount.html">1696</a></td>
            <td>1000</td>
            <td></td>
           </tr>
<tr>
            <td>Checkbox</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/32129/perf-test/Checkbox-mount_base.html">1687</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/32129/perf-test/Checkbox-mount.html">1702</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>CheckboxBase</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/32129/perf-test/CheckboxBase-mount_base.html">1466</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/32129/perf-test/CheckboxBase-mount.html">1462</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>ChoiceGroup</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/32129/perf-test/ChoiceGroup-mount_base.html">3016</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/32129/perf-test/ChoiceGroup-mount.html">2959</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>ComboBox</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/32129/perf-test/ComboBox-mount_base.html">672</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/32129/perf-test/ComboBox-mount.html">696</a></td>
            <td>1000</td>
            <td></td>
           </tr>
<tr>
            <td>CommandBar</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/32129/perf-test/CommandBar-mount_base.html">6540</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/32129/perf-test/CommandBar-mount.html">6542</a></td>
            <td>1000</td>
            <td></td>
           </tr>
<tr>
            <td>ContextualMenu</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/32129/perf-test/ContextualMenu-mount_base.html">11456</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/32129/perf-test/ContextualMenu-mount.html">11568</a></td>
            <td>1000</td>
            <td></td>
           </tr>
<tr>
            <td>DefaultButton</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/32129/perf-test/DefaultButton-mount_base.html">785</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/32129/perf-test/DefaultButton-mount.html">804</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>DetailsRow</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/32129/perf-test/DetailsRow-mount_base.html">2220</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/32129/perf-test/DetailsRow-mount.html">2275</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>DetailsRowFast</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/32129/perf-test/DetailsRowFast-mount_base.html">2195</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/32129/perf-test/DetailsRowFast-mount.html">2210</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>DetailsRowNoStyles</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/32129/perf-test/DetailsRowNoStyles-mount_base.html">2050</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/32129/perf-test/DetailsRowNoStyles-mount.html">2035</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Dialog</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/32129/perf-test/Dialog-mount_base.html">2720</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/32129/perf-test/Dialog-mount.html">2900</a></td>
            <td>1000</td>
            <td></td>
           </tr>
<tr>
            <td>DocumentCardTitle</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/32129/perf-test/DocumentCardTitle-mount_base.html">244</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/32129/perf-test/DocumentCardTitle-mount.html">235</a></td>
            <td>1000</td>
            <td></td>
           </tr>
<tr>
            <td>Dropdown</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/32129/perf-test/Dropdown-mount_base.html">2003</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/32129/perf-test/Dropdown-mount.html">2020</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>FocusTrapZone</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/32129/perf-test/FocusTrapZone-mount_base.html">1129</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/32129/perf-test/FocusTrapZone-mount.html">1133</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>FocusZone</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/32129/perf-test/FocusZone-mount_base.html">1126</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/32129/perf-test/FocusZone-mount.html">1089</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>GroupedList</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/32129/perf-test/GroupedList-mount_base.html">42296</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/32129/perf-test/GroupedList-mount.html">42341</a></td>
            <td>2</td>
            <td></td>
           </tr>
<tr>
            <td>GroupedList</td>
            <td>virtual-rerender</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/32129/perf-test/GroupedList-virtual-rerender_base.html">20381</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/32129/perf-test/GroupedList-virtual-rerender.html">20243</a></td>
            <td>2</td>
            <td></td>
           </tr>
<tr>
            <td>GroupedList</td>
            <td>virtual-rerender-with-unmount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/32129/perf-test/GroupedList-virtual-rerender-with-unmount_base.html">52585</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/32129/perf-test/GroupedList-virtual-rerender-with-unmount.html">52215</a></td>
            <td>2</td>
            <td></td>
           </tr>
<tr>
            <td>GroupedListV2</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/32129/perf-test/GroupedListV2-mount_base.html">227</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/32129/perf-test/GroupedListV2-mount.html">238</a></td>
            <td>2</td>
            <td></td>
           </tr>
<tr>
            <td>GroupedListV2</td>
            <td>virtual-rerender</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/32129/perf-test/GroupedListV2-virtual-rerender_base.html">210</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/32129/perf-test/GroupedListV2-virtual-rerender.html">218</a></td>
            <td>2</td>
            <td></td>
           </tr>
<tr>
            <td>GroupedListV2</td>
            <td>virtual-rerender-with-unmount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/32129/perf-test/GroupedListV2-virtual-rerender-with-unmount_base.html">231</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/32129/perf-test/GroupedListV2-virtual-rerender-with-unmount.html">228</a></td>
            <td>2</td>
            <td></td>
           </tr>
<tr>
            <td>IconButton</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/32129/perf-test/IconButton-mount_base.html">1132</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/32129/perf-test/IconButton-mount.html">1155</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Label</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/32129/perf-test/Label-mount_base.html">336</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/32129/perf-test/Label-mount.html">362</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Layer</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/32129/perf-test/Layer-mount_base.html">2753</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/32129/perf-test/Layer-mount.html">2781</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Link</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/32129/perf-test/Link-mount_base.html">397</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/32129/perf-test/Link-mount.html">395</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>MenuButton</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/32129/perf-test/MenuButton-mount_base.html">996</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/32129/perf-test/MenuButton-mount.html">993</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>MessageBar</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/32129/perf-test/MessageBar-mount_base.html">21208</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/32129/perf-test/MessageBar-mount.html">21292</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Nav</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/32129/perf-test/Nav-mount_base.html">2031</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/32129/perf-test/Nav-mount.html">2094</a></td>
            <td>1000</td>
            <td></td>
           </tr>
<tr>
            <td>OverflowSet</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/32129/perf-test/OverflowSet-mount_base.html">781</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/32129/perf-test/OverflowSet-mount.html">817</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Panel</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/32129/perf-test/Panel-mount_base.html">1840</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/32129/perf-test/Panel-mount.html">1856</a></td>
            <td>1000</td>
            <td></td>
           </tr>
<tr>
            <td>Persona</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/32129/perf-test/Persona-mount_base.html">743</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/32129/perf-test/Persona-mount.html">744</a></td>
            <td>1000</td>
            <td></td>
           </tr>
<tr>
            <td>Pivot</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/32129/perf-test/Pivot-mount_base.html">929</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/32129/perf-test/Pivot-mount.html">905</a></td>
            <td>1000</td>
            <td></td>
           </tr>
<tr>
            <td>PrimaryButton</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/32129/perf-test/PrimaryButton-mount_base.html">952</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/32129/perf-test/PrimaryButton-mount.html">946</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Rating</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/32129/perf-test/Rating-mount_base.html">4752</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/32129/perf-test/Rating-mount.html">4791</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>SearchBox</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/32129/perf-test/SearchBox-mount_base.html">928</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/32129/perf-test/SearchBox-mount.html">924</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Shimmer</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/32129/perf-test/Shimmer-mount_base.html">1895</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/32129/perf-test/Shimmer-mount.html">1896</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Slider</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/32129/perf-test/Slider-mount_base.html">1357</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/32129/perf-test/Slider-mount.html">1396</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>SpinButton</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/32129/perf-test/SpinButton-mount_base.html">3043</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/32129/perf-test/SpinButton-mount.html">3003</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Spinner</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/32129/perf-test/Spinner-mount_base.html">407</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/32129/perf-test/Spinner-mount.html">405</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>SplitButton</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/32129/perf-test/SplitButton-mount_base.html">1881</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/32129/perf-test/SplitButton-mount.html">1941</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Stack</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/32129/perf-test/Stack-mount_base.html">421</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/32129/perf-test/Stack-mount.html">416</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>StackWithIntrinsicChildren</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/32129/perf-test/StackWithIntrinsicChildren-mount_base.html">892</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/32129/perf-test/StackWithIntrinsicChildren-mount.html">867</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>StackWithTextChildren</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/32129/perf-test/StackWithTextChildren-mount_base.html">2761</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/32129/perf-test/StackWithTextChildren-mount.html">2833</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>SwatchColorPicker</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/32129/perf-test/SwatchColorPicker-mount_base.html">6466</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/32129/perf-test/SwatchColorPicker-mount.html">6418</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>TagPicker</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/32129/perf-test/TagPicker-mount_base.html">1486</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/32129/perf-test/TagPicker-mount.html">1458</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Text</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/32129/perf-test/Text-mount_base.html">387</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/32129/perf-test/Text-mount.html">380</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>TextField</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/32129/perf-test/TextField-mount_base.html">966</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/32129/perf-test/TextField-mount.html">953</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>ThemeProvider</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/32129/perf-test/ThemeProvider-mount_base.html">873</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/32129/perf-test/ThemeProvider-mount.html">853</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>ThemeProvider</td>
            <td>virtual-rerender</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/32129/perf-test/ThemeProvider-virtual-rerender_base.html">610</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/32129/perf-test/ThemeProvider-virtual-rerender.html">593</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>ThemeProvider</td>
            <td>virtual-rerender-with-unmount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/32129/perf-test/ThemeProvider-virtual-rerender-with-unmount_base.html">1306</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/32129/perf-test/ThemeProvider-virtual-rerender-with-unmount.html">1328</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Toggle</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/32129/perf-test/Toggle-mount_base.html">617</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/32129/perf-test/Toggle-mount.html">617</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>buttonNative</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/32129/perf-test/buttonNative-mount_base.html">195</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/32129/perf-test/buttonNative-mount.html">186</a></td>
            <td>5000</td>
            <td></td>
           </tr></table></p></details>

<div id="perfComment9423"/>

fabricteam commented on apps/vr-tests-web-components/package.json:
```diff
@@ -12,9 +12,6 @@
     "type-check": "tsc -p . --baseUrl . --noEmit",
```
## 🕵🏾‍♀️ [<b data-uid='vrt-gist'></b> visual regressions to review in the fluentuiv9 Visual Regression Report](https://staticpagerelease.z16.web.core.windows.net/azurestaticapps/vr-approval/index.html?organization=uifabric&project=fabricpublic&prId=32129&commitId=1a9957abf68b01a502f9e7421ad676b3cf0fd746&env=prod&clientType=FLUENTUI)


  <details><summary><b>Avatar Converged</b> 1 screenshots</summary>

|Image Name|Diff(in Pixels)| Image Type
|-|-|-|
|Avatar Converged.badgeMask.normal.chromium.png|3| Changed|

</details>




<div id="vrtCommentfluent-ui_VRT_Pipeline_v9"/>

Hotell reviewed (commented):
> package are incorrectly including reference tags (/// <reference types="web" />), which creates issues for projects which use @fluentui/web-components as a dev dependency.

this seems to me like issue in user land. folks should use `skipLibCheck:true` in their TS configs

> the upgrade from TypeScript 4 to 5 exposed a tsconfig configuration error with the moduleResolution field.

what kind of error ?  AFAIR `moduleResolution` node16 follows same resolution as 
 nodeNext


---

I already provided review on this approach some time ago https://github.com/microsoft/fluentui/pull/31807#pullrequestreview-2138273703

this introduces another complex setup to our repo which should be avoided. 

if you are ok with shipping leaking global types without proper type definitions to your users I'd suggest to write a simple script that removes those type references after build step from `.d.ts` files.

Note: it seems `menu` and `menu-item` expose these `web` types to public API surface

quickly checking the implementation code indeed those types do leak:
![image](https://github.com/user-attachments/assets/97d306bb-7501-49f0-89a4-d84e1c3d3752)

**Summary:**
- we should not land these monorepo setup changes
- there are multiple patterns how to go around these limitations

davatron5000 commented:
Created issue for leaking types #32194

radium-v commented:
Hi @Hotell,

Using `skipLibCheck` to resolve conflicts with the TypeScript standard library is discouraged, as stated in the documentation for [skipLibCheck](https://www.typescriptlang.org/tsconfig/#skipLibCheck). It's recommended to use lib replacement.

I reverted the `moduleResolution` changes shortly after creating the PR (I missed a reference which I have now reverted as well, and updated the PR comment to remove that item). It's been a recurring issue that the `module` and `moduleResolution` fields show misconfiguration errors in my editor. I'm sorry for any confusion there.

We're not okay with shipping leaking types, which is what this PR is specifically intended to address.

Writing a custom script to do what a few simple, standard, and recommended configuration adjustments already do, in a built-in way, doesn't make sense to me.

radium-v commented on package.json:
```diff
@@ -365,7 +365,8 @@
     ],
     "nohoist": [
-      "@fluentui/web-components/@storybook/html"
+      "@fluentui/web-components/@storybook/html",
+      "@fluentui/web-components/@typescript/lib-dom"
```
The `@typescript/lib-dom` package is included in `nohoist`, since `@types/web` breaks in packages which target ES5. This ensures that this types override is scoped only to the web components package.

https://github.com/microsoft/TypeScript-DOM-Lib-Generator?tab=readme-ov-file#typeslib-minimum-target
![image](https://github.com/user-attachments/assets/dde75ec3-98a8-4c7d-813e-129d5e591ad7)

chrisdholt replied:
Thanks for this - makes sense!

radium-v commented:
I reduced the changes down to only those related to the lib replacement, and fixing the type error in a utility class.

davatron5000 commented:
I started looking into the leaking types issue and couldn't quite replicate your red squiggles, but after talking with @radium-v it seems that this PR would help fix the issue with `ToggleEvent` (a new but valid web platform event object type). 

As mentioned in the initial issue...

- `/// <reference types="web" />` is causing some issues in downstream consumers.
- Downstream consumers could add `@types/web` to their tsconfigs, but if they're using newer versions of TS (5.5) they'll run into duplicate type errors.
- We can't tell consumers to use `skipLibCheck`
- The [`skipLibCheck` documentation](https://www.typescriptlang.org/tsconfig/#skipLibCheck) seems to push people away from using `skipLibCheck` to solve this usecase.
- The `skipLibCheck` documentation is pushing us towards using the lib replacement strategy
- The [`@types/web` documentation](https://www.npmjs.com/package/@types/web) is also pushing us towards the lib replacement strategy

Could explain more about the concerns you have for the lib-replacement strategy landing in the web-components package? I'd like to fully understand the problems and perceived complexity.

chrisdholt commented on apps/vr-tests-web-components/package.json:
```diff
@@ -12,9 +12,6 @@
     "type-check": "tsc -p . --baseUrl . --noEmit",
     "test-vr": "storywright  --browsers chromium --url dist/storybook --destpath dist/screenshots --waitTimeScreenshot 500 --concurrency 4 --headless true"
   },
-  "devDependencies": {
-    "@types/web": "^0.0.142"
-  },
```
@radium-v confirming this was removed because it's not needed.

chrisdholt commented on apps/vr-tests-web-components/tsconfig.json:
```diff
@@ -8,7 +8,7 @@
     "resolveJsonModule": true,
     "allowJs": true,
     "jsx": "react",
-    "types": ["jest", "node", "web"]
```
confirming this is removed as unnecessary to the VR package

chrisdholt reviewed (approved):
Just a couple questions to confirm

Hotell commented:
**folks, quick tip:** if you wanna another review please make sure to hit the refresh button, otherwise reviewers will never get notification that there has been some movement. ty ! 
<img width="362" alt="image" src="https://github.com/user-attachments/assets/f57357a1-3c19-4f9e-a4b2-1fcd69c3946f">

davatron5000 commented:
I can "fix" the `ToggleEvent` issue by rewriting the `toggleHandler` to not use the appropriate type and just use `Event`. It sort of defeats the entire purpose of TypeScript, but hopefully that unblocks us there.
 
That said, the web-components project will always be a bit ahead of the curve on web platform feature we use, so the `ToggleEvent` issue a convenient example of where we're ahead of the current monorepo version of TS in terms of supported web APIs.

Per our hours of investigation into the issue...

- The `types: ["web"]` method is a mostly undocumented way of extending web types for TS <=4.4. ([see docs](https://github.com/microsoft/TypeScript-DOM-lib-generator/blob/main/deploy/readmes/web.md))
  - The TS <=4.4 docs are hidden in a dropdown and don't mention the `types` array technique, but instead recommend using `lib: ['SomeESTarget']` or removing `lib: ["dom"]` if `lib` already exists.
- Since TS 4.5+, TypeScript and `@types/web` recommends using the lib-replacement method for extending web types.

We did some experiments yesterday and made the exact same demo repro you posted today and verified the `tsc` compiler couldn't build in the downstream project. Unfortunately, post-processing removing the `/// references` pragma from all the `.d.ts` files only half-solved our problem. Downstream consumers were getting duplicate type errors for all types (web-components' types/web + local-repo's lib-dom types). The easiest way for us to resolve that issue was to use lib-replacement. @radium-v and I can work up a demo/video if that would help demonstrate the issue.

With lib-replacement, the leaking types issue only surfaces if a downstream consumer tries to overrride/extend a function that uses newer types. At that point they are using newer types and would need to upgrade their lib-dom types to web-types. For downstream consumers in that situation (TS <= 5.4) we would not recommend `skipLibCheck` (per the docs) and recommend the lib-replacement strategy (per the docs). So for us, all paths lead us to lib-replacement.

If the web-component package's use of lib-replacement for web types with `nohoist` guards in effect breaks other systems in the monorepo in some way, please let us know. But we've done our due diligence here and are trying abide in the one-version policy while not impacting other projects but still allowing us to leverage new web technologies.

chrisdholt commented:
Closing this in favor of https://github.com/microsoft/fluentui/pull/32308
